File: 0001. Two Sum/java/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        var diff = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            var value = nums[i];
            var found = target - nums[i];

            if (diff.containKey(found)) {
                return new int[] {diff.get(found), i};
            }
            diff.put(value, i);
        }
        return new int[] {-1, -1};
    }
}

File: 0001. Two Sum/kotlin/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val diff = HashMap<Int, Int>()

        for (i in 0 until nums.count()) {
            val value = nums[i]
            val found = target - value

            diff.get(found)?.let {
                return intArrayOf(it, i)
            }
            diff.put(value, i)
        }
        return intArrayOf(-1, -1)
    }
}
